### PR TITLE
Updates IPSEC method for 4.15

### PR DESCRIPTION
**Pull request by chentex**
## Type of change

- [x] Refactor
- [ ] New feature
- [x] Bug fix
- [ ] Optimization
- [ ] Documentation Update

## Description

Updates IPSEC method for 4.15
This should also avoid the errors that have been appearing in older versions

## Related Tickets & Documents

- Related Issue #
- Closes #

## Checklist before requesting a review

- [ ] I have performed a self-review of my code.
- [ ] If it is a core feature, I have added thorough tests.

## Testing
- Please describe the System Under Test.
- Please provide detailed steps to perform tests related to this code change.
- How were the fix/results from this change verified? Please provide relevant screenshots or results.

### CODE REVIEW

**Comment by chentex** [comment]
Added the collection of `ipsecMode`

**Comment by vishnuchalla** [comment]
Can you update your fork with latest mainstream changes and merge them into your fork and update the PR please? I doubt if thats the reason our CI test is failing.

File: utils/index.sh [hunk 1]
```diff
@@ -71,23 +71,27 @@ setup(){
...
-
-    # Check if the pod has the container nbdb, if it is OVNIC it won't
-    # If it is a OVN the command will succed and the result will be stored in ipsec
-    if result=$(oc -n openshift-ovn-kubernetes -c nbdb rsh $ovn_pod ovn-nbctl --no-leader-only get nb_global . ipsec); then
-        if [[ $result == *"true"* ]]; then
-            ipsec=true
+    ipsecMode="Disabled"
+    if result=$(oc get networks.operator.openshift.io cluster -o=jsonpath='{.spec.defaultNetwork.ovnKubernetesConfig.ipsecConfig.mode}'); then
+        # If $result is empty, it is version older than 4.15
+        # We need to chekc a level above in the jsonpath
```
Comment (vishnuchalla) [inline utils/index.sh:78]: typo on `chekc`?

File: utils/index.sh [hunk 2]
```diff
@@ -71,23 +71,27 @@ setup(){
...
+            if deprecatedresult=$(oc get networks.operator.openshift.io cluster -o=jsonpath='{.spec.defaultNetwork.ovnKubernetesConfig.ipsecConfig}'); then
+                if [[ ! -z $deprecatedresult ]]; then
+                    ipsec=true
+                    ipsecMode="Full"
+                fi
+            fi
+        else
+            # No matter if enabled and then disabled or
+            # Disabled by default
+            # this field is always shows Disabled when no IPSec
```
Comment (vishnuchalla) [inline utils/index.sh:90]: rephrase please?

**Comment by vishnuchalla** [comment]
Except for the nitpicks assuming its tested in a cluster, looks good to me.

File: utils/index.sh (hunk 2 above)
Comment (chentex) [reply utils/index.sh:90]: Any suggestions, on how to explain the situation

Comment (vishnuchalla) [reply utils/index.sh:90]:
```suggestion
            # No matter if Enabled and then Disabled or Disabled by default, this field always shows Disabled when no IPSec.
```

**Pull request merged**